ui/admin/app/lib/service/api/modelProviderApiService.ts

```diff
@@ -1,16 +1,90 @@
-import { ModelProvider } from "~/lib/model/models";
+import { ModelProvider, ModelProviderConfig } from "~/lib/model/modelProviders";
 import { ApiRoutes } from "~/lib/routers/apiRoutes";
 import { request } from "~/lib/service/api/primitives";
 
-async function getModelProviders() {
-    const { data } = await request<{ items?: ModelProvider[] }>({
+const getModelProviders = async () => {
+    const res = await request<{ items: ModelProvider[] }>({
         url: ApiRoutes.modelProviders.getModelProviders().url,
+        errorMessage: "Failed to get supported model providers.",
     });
 
-    return data.items ?? [];
-}
-getModelProviders.key = () => ({
-    url: ApiRoutes.modelProviders.getModelProviders().path,
-});
+    return res.data.items ?? ([] as ModelProvider[]);
+};
+getModelProviders.key = () =>
+    ({ url: ApiRoutes.modelProviders.getModelProviders().path }) as const;
 
-export const ModelProviderApiService = { getModelProviders };
+const getModelProviderById = async (modelProviderKey: string) => {
+    const res = await request<ModelProvider>({
+        url: ApiRoutes.modelProviders.getModelProviderById(modelProviderKey)
+            .url,
+        method: "GET",
+        errorMessage:
+            "Failed to update configuration values on the requested modal provider.",
+    });
+
+    return res.data;
+};
+getModelProviderById.key = (modelProviderId?: string) => {
+    if (!modelProviderId) return null;
+
+    return {
+        url: ApiRoutes.modelProviders.getModelProviderById(modelProviderId)
+            .path,
+        modelProviderId,
+    };
+};
+
+const configureModelProviderById = async (
+    modelProviderKey: string,
+    modelProviderConfig: ModelProviderConfig
+) => {
+    const res = await request<ModelProvider>({
+        url: ApiRoutes.modelProviders.configureModelProviderById(
+            modelProviderKey
+        ).url,
+        method: "POST",
+        data: modelProviderConfig,
+        errorMessage:
+            "Failed to update configuration values on the requested modal provider.",
+    });
+
+    return res.data;
+};
+configureModelProviderById.key = (modelProviderId?: string) => {
+    if (!modelProviderId) return null;
+
+    return {
+        url: ApiRoutes.modelProviders.configureModelProviderById(
+            modelProviderId
+        ).path,
+        modelProviderId,
+    };
+};
```

Comment:
nit: most apis that mutate data should not be triggered via SWR and therefore don't require a key

Reply:
Thank you! Will remove that